core/server/data/migrations/versions/4.35/2022-02-02-13-10-transform-specific-tiers-default-content-visibility.js

```diff
@@ -0,0 +1,114 @@
+const logging = require('@tryghost/logging');
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Checking default_content_visibility for specific tiers');
+
+        const settings = await knex('settings')
+            .select()
+            .whereIn('key', ['default_content_visibility', 'default_content_visibility_tiers']);
+        const contentVisibilitySetting = settings.find(d => d.key === 'default_content_visibility');
+        const visibilityTiersSetting = settings.find(d => d.key === 'default_content_visibility_tiers');
+        if (!contentVisibilitySetting || !contentVisibilitySetting.value) {
+            logging.warn('No default_content_visibility setting found.');
+            return;
+        }
+
+        if (!visibilityTiersSetting) {
+            logging.warn('No default_content_visibility_tiers setting found.');
+            return;
+        }
+
+        const contentVisibility = contentVisibilitySetting.value;
+        // Transform visibility to tiers when stored as nql string
+        if (!['public', 'members', 'paid'].includes(contentVisibility)) {
+            const isValidProductNqlFilter = /^(?:product:[\w-]+,?)+$/.test(contentVisibility);
+            let newVisibilityValue = 'paid';
+            let newVisibilityTiersValue = [];
+            if (isValidProductNqlFilter) {
+                newVisibilityValue = 'tiers';
+                // fetch product slugs from nql filter
+                const productSlugs = contentVisibility.split(',').map((segment) => {
+                    return segment.replace('product:', '');
+                });
+
+                // get product ids for slugs
+                const products = await knex('products')
+                    .select('id')
+                    .whereIn('slug', productSlugs);
+                newVisibilityTiersValue = products.map((product) => {
+                    return product.id;
+                });
+            }
+
+            logging.info(`Updating default_content_visibility to tiers`);
```

Comment:
This isn't strictly true, it should probably be 

```js
logging.info(`Updating default_content_visibility to ${newVisibilityValue}`);
```

Reply:
✅